corehq/apps/cloudcare/utils.py

```diff
@@ -31,3 +35,26 @@ def webapps_module_case_form(domain, app_id, module_id, case_id, form_id):
 
 def webapps_module(domain, app_id, module_id):
     return _webapps_url(domain, app_id, selections=[module_id])
+
+
+def should_restrict_web_apps_usage(domain):
```

Comment:
It just occurred to me that this would be a good target for caching.  It'll only fire if `ALLOW_WEB_APPS_RESTRICTION` is disabled, so it mostly won't matter, but when that _is_ enabled, iterating through all apps report configs will take a decent amount of time.